docs/docs/code/python/http-requests/README.md

```diff
@@ -39,24 +83,215 @@ def handler(pd: "pipedream"):
+
+```python
+import requests
+
+def handler(pd: "pipedream"):
+  # Download the webpage HTML file to /tmp
+  with requests.get("https://example.com", stream=True) as response:
+      # Check if the request was successful
+      response.raise_for_status()
+
+      # Open the new file /tmp/file.html in binary write mode
+      with open("/tmp/file.html", "wb") as file:
+          for chunk in response.iter_content(chunk_size=8192):
+              # Write the chunk to file
+              file.write(chunk)
+```
+
+## Uploading a file from the `/tmp` directory
+
+This example shows you how to make a `multipart/form-data` request with a file as a form part. You can store and read any files from [the `/tmp` directory](/code/python/working-with-files/#the-tmp-directory).
+
+This can be especially helpful for uploading large files: it streams the file from disk, minimizing the memory the workflow uses when uploading the file.
+
+```python
+import requests
+from requests_toolbelt.multipart.encoder import MultipartEncoder
+
+def handler(pd: "pipedream"):
+  m = MultipartEncoder(fields={
+    'file': ('filename', open('/tmp/file.pdf', 'rb'), 'application/pdf')
+    })
+
+  r = requests.post("https://example.com", data=m,
+                  headers={'Content-Type': m.content_type})
+```
+
+## IP addresses for HTTP requests made from Pipedream workflows
+
+By default, [HTTP requests made from Pipedream can come from a large range of IP addresses](/privacy-and-security/#hosting-details). **If you need to restrict the IP addresses HTTP requests come from, you can [Use a Pipedream VPC](/workflows/vpc/) to route all outbound HTTP requests through a single IP address.**
+
+## Using an HTTP proxy to proxy requests through another host
+
+By default, HTTP requests made from Pipedream can come from a range of IP addresses. **If you need to make requests from a single IP address, you can route traffic through an HTTP proxy**:
+
+```python
+import requests
+
+def handler(pd: "pipedream"):
+  user = "USERNAME" # Replace with your HTTP proxy username
+  password = "PASSWORD" # Replace with your HTTP proxy password
+  host = "10.10.1.10" # Replace with the HTTP proxy URL
+  port = 1080 # Replace with the port of the HTTP proxy
+  proxies = {
+    "https": f"http://{user}:{password}@{host}:{port}",
+  }
+
+  r = requests.request("GET", "https://example.com", proxies=proxies)
+```
+
+## Paginating API requests
+
+When you fetch data from an API, the API may return records in "pages". For example, if you're trying to fetch a list of 1,000 records, the API might return those in groups of 100 items.
+
+Different APIs paginate data in different ways. You'll need to consult the docs of your API provider to see how they suggest you paginate through records.
```

Comment:
This is copied from: https://github.com/PipedreamHQ/pipedream/blob/284df1945cd4a94c30ed929794f9a23eb44e5bf9/docs/docs/code/nodejs/http-requests/README.md#L622

Should this kind of content be de-duplicated in the future?

Reply:
This content will be removed in the future once we fully migrate users onto VPCs that currently use the managed proxy service.